test/controllers/sections/items/ItemProducerSizeControllerSpec.scala

```diff
@@ -0,0 +1,212 @@
+      messagesApi = messagesApi,
+      userAnswersService = mockUserAnswersService,
+      navigator = new FakeItemsNavigator(testOnwardRoute),
+      auth = fakeAuthAction,
+      getData = new FakeDataRetrievalAction(userAnswers, Some(testMinTraderKnownFacts)),
+      requireData = dataRequiredAction,
+      userAllowList = fakeUserAllowListAction,
+      formProvider = formProvider,
+      controllerComponents = Helpers.stubMessagesControllerComponents(),
+      view = view,
+      timeMachine = TimeMachine,
+      cnCodeInformationService = mockGetCnCodeInformationService
+    )
+
+    def itemProducerSizeRoute(idx: Index): String = routes.ItemProducerSizeController.onPageLoad(testErn, testDraftId, idx, NormalMode).url
+    def onSubmitAction(idx: Index): Call = routes.ItemProducerSizeController.onSubmit(testErn, testDraftId, idx, NormalMode)
+  }
+
+
+  "ItemProducerSize Controller" - {
+
+    "for a GET onPageLoad" - {
+
+      "must return OK and the correct view for a GET with a goodstype set" in new Setup(Some(
+        emptyUserAnswers.set(ItemExciseProductCodePage(0), Wine.code)
+      )) {
+
+        val request: FakeRequest[AnyContentAsEmpty.type] = FakeRequest(GET, itemProducerSizeRoute(0))
+
+        val result: Future[Result] = TestController.onPageLoad(testErn, testDraftId, 0, NormalMode)(request)
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(
+          form = form,
+          onSubmitAction = onSubmitAction(0),
+          goodsType = Wine,
+          startYear = "2022",
+          endYear = "2023"
+        )(dataRequest(request), messages(request)).toString
+      }
+
+      "must return OK and the correct view for a GET when the current date is January" in new Setup(
+        userAnswers = Some(emptyUserAnswers.set(ItemExciseProductCodePage(0), Wine.code)),
+        monthValue = 1
+      ) {
+
+        val request = FakeRequest(GET, itemProducerSizeRoute(0))
+
+        val result: Future[Result] = TestController.onPageLoad(testErn, testDraftId, 0, NormalMode)(request)
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(
+          form = form,
+          onSubmitAction = onSubmitAction(0),
+          goodsType = Wine,
+          startYear = "2021",
+          endYear = "2022"
+        )(dataRequest(request), messages(request)).toString
+      }
+
+      "must populate the view correctly on a GET when the question has previously been answered" in new Setup(Some(
+        emptyUserAnswers
+          .set(ItemExciseProductCodePage(0), Wine.code)
+          .set(ItemProducerSizePage(0), 1)
+      )) {
+
+        val request = FakeRequest(GET, itemProducerSizeRoute(0))
+
+        val result: Future[Result] = TestController.onPageLoad(testErn, testDraftId, 0, NormalMode)(request)
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(
+          form = form.fill(1),
+          onSubmitAction = onSubmitAction(0),
```

Comment:
```suggestion
          onSubmitAction = onSubmitAction(testIndex1),
```

Reply:
I don't think we need this since we have the implicit `intToIndex` def